pkg/services/guardian/guardian.go

```diff
@@ -43,17 +43,23 @@ type dashboardGuardianImpl struct {
 	teams  []*models.TeamDTO
 	log    log.Logger
 	ctx    context.Context
+	store  sqlstore.Store
 }
 
 // New factory for creating a new dashboard guardian instance
 // When using access control this function is replaced on startup and the AccessControlDashboardGuardian is returned
 var New = func(ctx context.Context, dashId int64, orgId int64, user *models.SignedInUser) DashboardGuardian {
+	panic("no guardian factory implementation provided")
```

Comment:
:eyes: There are a lot of users of this function, I get why you didn't refactor this call with an error signature. All things taken together, I think this is an improvement even if it introduces a new somewhat scary panic.

Reply:
I expected 3 use cases:

* Regular Grafana backend would be using ProvideService and there will be no panic, since guardians would be initialised properly. If for some reason Wire gets broken - a panic would occur highlighting it.
* Integration tests, where ProvideService is still in use, so broken Wire results in a panic.
* Unit tests, where guardian is often initialised manually and panic helps a lot in this case to find broken tests.